### PR TITLE
Add Cypress Rancher import Harvester test cases 

### PR DESCRIPTION
**Pull request by TachunLin**
This the initial commit of Rancher import Harvester for Cypress automation test  
For better review and update, split the original Cypress Rancher integration PR https://github.com/harvester/tests/pull/502 into smaller part to proceed. 

### Test Cases 
1. Prepare Harvester Image
2. Prepare Harvester VLAN network
3. Rancher import Harvester
4. Harvester import Rancher
5. Check Harvester Cluster Statu

### Purpose 
Develop Cypress test for Harvester Rancher integration related test cases. 
Plan to fulfill the requirement to automated basic regression and KDM release test.  

### Development Environment 
The development based on the following SUT version and UI

* Rancher version: v2.7.5
* Harvester version: v1.2.0-rc4 
* Cypress version: 10.8

### Prepare Test Environment 
We can prepare the Rancher integration test environment in the following manner  to provision Harvester with Rancher, 
1. Use vagrant ipxe-example pull request [45](https://github.com/harvester/ipxe-examples/pull/45), set
  - harvester_network_config.offline: `false` 
  - rancher_config.run_single_node_rancher: `true`  
  - rancher_config.run_single_node_air_gapped_rancher: `false`

2. Create separate VMs, one for Harvester and another for Rancher, make sure the network connectivity
3. Install Harvester and Rancher on bare machines and ensure the connectivity

### About Rancher First Time Login
_**Please be noted**_
In every test, while login Rancher, it will automatically check whether it's the first time login 
The test function `rancher.rancherLogin()` support bootstrap login in the following scenarios.

1. Use vagrant ipxe-example to prepare Harvester and Rancher and install `sshpass` on your local test runner environment 
2. You can also use regular VM or bare machine while you already setup the following 
 - ssh root permission to Rancher host VM or bare machine 
 - Install `sshpass` on your local test runner environment 

**_If your environment did not fulfill above requirement, we would suggest you finishing the Rancher bootstrap on your own before running the rancher integration test_**

### Cypress setup
1. Execute `npm install`
3. Execute `npm ci` 
4. Rename `cypress.env.json.example` to `cypress.env.json` file, update and add the following content 
  ```
  "username": "admin",
  "password": "passwdpasswd",
  "baseUrl": "https://192.168.122.149", 
  "rancherUrl": "https://192.168.122.224",
  "rancher_user": "admin",
  "rancher_password": "passwdpasswd",
  "rancher_vm_user": "vagrant",  
  "rancher_vm_password": "vagrant",   
   "vagrant_pxe_path": "../ipxe-examples/vagrant-pxe-harvester",
  ```

### Check test data
You can check and modify all related rancher integration test data in `../cypress/fixtures/rancher.json`
The test data here is suitable for `Rancher v2.6.7` along with the current KDM release version 
Please be noted, on new KDM release, you may need to update these value accordingly. 

```
    "rke2_latest": "1.24.4",
    "rke2_stable": "1.23.10",
    "k3s_latest": "1.24.4",
    "k3s_stable": "1.23.10", 
    "rke1_latest": "1.24.4",
    "rke1_stable": "1.23.10",
```
And also need to check the image and ssh user name to fulfill what you used in the `Prepare Harvester Image` and `Prepare Harvester VLAN network` test case, here I use focal-server and default vlan1 

```
    "image": "focal-server",
    "network_name": "default/vlan1",
    "ssh_user": "ubuntu",
```

I use ubuntu 20.04 cloud focal by settings the image name and url in  `cypress.env.json`

```
"image":
  {
    "name": "openSUSE-Leap-15.3-3-DVD-x86_64-Build38.1-Media.iso",
    "url": "https://download.opensuse.org/distribution/leap/15.3/iso/openSUSE-Leap-15.3-3-DVD-x86_64-Build38.1-Media.iso"
  }
```

### Execute Test case
Execute `npx cypress open` or `npx cypress run` for the test 

###  Test case list 
1. Prepare Harvester Image (On Harvester)
2. Prepare Harvester VLAN network (On Harvester)
3. Rancher import Harvester 
4. Harvester import Rancher (On Harvester)
5. Check Harvester Cluster Status (On Rancher) 

More test case will be added later on other PR

### CODE REVIEW

File: cypress/pageobjects/rancher.po.ts [hunk 1]
```diff
@@ -0,0 +1,497 @@
...
+    private boostrap_page_radioSelectPW = ':nth-child(2) > .radio-container';
+    private boostrap_page_newPWInput = ':nth-child(5) > .password > .labeled-input > input';
+    private boostrap_page_newPWRepeat = '[style=""] > .labeled-input > input';
+    // private boostrap_page_checkAgreeEULA = '#checkbox-eula > .checkbox-container > .checkbox-custom';
+    private boostrap_page_checkAgreeEULA = '[data-testid="setup-agreement"] > .checkbox-container > .checkbox-custom';
+    private boostrap_page_confirmLogin = '.btn > span';
+    private home_page_mainMenu = '.menu';
+    private home_page_virtualManagement = ':nth-child(7) > .option > div';
+
+    private virtual_page_importButton = '.btn';
```
Comment (n313893254) [inline cypress/pageobjects/rancher.po.ts:49]:
```suggestion
    private virtual_page_importButton = '.actions > .btn';
```

**Review by n313893254 — CHANGES_REQUESTED** [review]
Minor fixes, Others LGTM

image

File: cypress/pageobjects/rancher.po.ts (hunk 1 above)
Comment (TachunLin) [reply cypress/pageobjects/rancher.po.ts:49]: Thanks for the check and reminder. 
I have updated the page object entry and tested can pass the entire Harvester test suite.

**Comment by TachunLin** [comment]
Verified can pass the following test cases on Rancher v2.7.6-rc2 with Harvester v1.1.2-rc5.
Thanks for the review. 

image

**Review by n313893254 — APPROVED** [review]
LGTM, Thanks!

**Pull request merged**